File: src-winui/src/GSBT.Core/Common/CatalogGameKeys.cs
using GSBT.Core.Models;

namespace GSBT.Core.Common;

/// <summary>
/// Canonical key for matching <see cref="GameRecord"/> to persisted catalog rows (<see cref="Models.SaveScanResult"/> names).
/// </summary>
public static class CatalogGameKeys
{
    /// <summary>
    /// Same naming rule as <c>SaveScanResult.Name</c> produced by <c>ScanService.ProcessSingleGame</c>.
    /// </summary>
    public static string FromDetectedGame(GameRecord game) =>
        game.CatalogDisplayName ?? GameDisplayName.CleanDisplayName(game.Name);
}

File: src-pyqt/core/catalog_game_keys.py
"""Canonical catalog keys for detected games (C# CatalogGameKeys parity)."""

from __future__ import annotations

import re


def clean_display_name(name: str) -> str:
    s = re.sub(r"[®™©]", "", name or "").strip()
    return re.sub(r"\s+", " ", s)


def catalog_key_from_detected_game(game: dict) -> str:
    return clean_display_name(str(game.get("name") or ""))
